pkg/cmd/cli/cmd/project.go

```diff
@@ -105,6 +105,15 @@ func (o *ProjectOptions) Complete(f *clientcmd.Factory, args []string, out io.Wr
 
 	o.ClientConfig, err = f.ClientConfig()
 	if err != nil {
+		// if the argument for o.ProjectName passed by the user is a context name,
+		// prevent local context-switching from failing due to an unreachable
+		// server or an unfetchable ClientConfig.
+		if _, exists := o.GetContextFromName(o.ProjectName); exists {
+			o.Config.CurrentContext = o.ProjectName
+			if err := kclientcmd.ModifyConfig(o.PathOptions, o.Config, true); err != nil {
```

Comment:
> Hrm, is this error typed? If we can detect the exact reason I think we should bypass it and switch cleanly, without reporting the error

Sounds good, we could check to see if error is [errors.IsNotFound](https://github.com/openshift/origin/blob/master/vendor/k8s.io/kubernetes/pkg/api/errors/errors.go#L363) or [errors.IsInternalError](https://github.com/openshift/origin/blob/master/vendor/k8s.io/kubernetes/pkg/api/errors/errors.go#L412), and skip to switching the context if this is the case. However, the last [few messages that are outputted to a user after successfully switching contexts](https://github.com/openshift/origin/blob/master/pkg/cmd/cli/cmd/project.go#L261-L279) are usually created with the host of the new context, using the `ClientConfig` that caused this error to begin with:

```
$ oc project sample-app-2/api-preview-openshift-com:443/juanvallejo
Already on project "sample-app-2" on server "https://api.preview.openshift.com:443".
```

Maybe in the case of this error happening, we could modify the message to use the new "context name" instead of its hostname?